corehq/ex-submodules/pillowtop/feed/couch.py

```diff
@@ -47,8 +47,12 @@ def couch_db(self):
 
 
 def change_from_couch_row(couch_change, document_store=None):
+    try:
+        change_id = couch_change['id']
+    except KeyError as err:
+        raise ValueError(f'ID not found in change {couch_change!r}') from err
```

Comment:
For example: 
```
Traceback (most recent call last):
  File "/srv/share/src/dimagi/commcare-hq/corehq/ex-submodules/pillowtop/feed/couch.py", line 51, in change_from_couch_row
    change_id = couch_change['id']
KeyError: 'id'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/srv/share/src/dimagi/commcare-hq/corehq/ex-submodules/pillowtop/pillow/interface.py", line 108, in run
    self.process_changes(since=self.get_last_checkpoint_sequence(), forever=True)
  File "/srv/share/src/dimagi/commcare-hq/corehq/ex-submodules/pillowtop/pillow/interface.py", line 156, in process_changes
    for change in self.get_change_feed().iter_changes(since=since or None, forever=forever):
  File "/srv/share/src/dimagi/commcare-hq/corehq/ex-submodules/pillowtop/feed/couch.py", line 30, in iter_changes
    change = change_from_couch_row(couch_change, document_store=self._document_store)
  File "/srv/share/src/dimagi/commcare-hq/corehq/ex-submodules/pillowtop/feed/couch.py", line 53, in change_from_couch_row
    raise ValueError(f'ID not found in change {couch_change!r}') from err
ValueError: ID not found in change {'_id': '026c0353a410eef9bed62b12840151b6', '_rev': '14-4d8c33ba14354ccb82282feae7576553', 'program_id': None, 'last_password_set': '1900-01-01T00:00:00.000000Z', 'fcm_device_token': None, 'atypical_user': False, 'doc_type': 'WebUser', 'base_doc': 'CouchUser', 'device_ids': [], 'devices': [], 'last_device': {'device_id': None, 'last_used': None, 'commcare_version': None, 'app_meta': [], 'doc_type': 'DeviceIdLastUsed'}, 'phone_numbers': [], 'created_on': '2020-02-27T10:42:43.004006Z', 'last_modified': '2020-03-04T00:30:36.905756Z', 'status': None, 'language': None, 'subscribed_to_commcare_users': False, 'announcements_seen': [], 'user_data': {'commcare_project': None}, 'location_id': None, 'assigned_location_ids': [], 'has_built_app': True, 'analytics_enabled': True, 'two_factor_auth_disabled_until': None, 'login_attempts': 0, 'attempt_date': None, 'reporting_metadata': {'last_submissions': [{'app_id': '026c0353a410eef9bed62b128401cc36', 'submission_date': '2020-03-03T17:03:58.706160Z', 'build_id': None, 'device_id': 'Formplayer', 'build_version': None, 'commcare_version': None, 'doc_type': 'LastSubmission'}, {'app_id': None, 'submission_date': '2020-02-27T12:53:38.486688Z', 'build_id': None, 'device_id': 'Formplayer', 'build_version': None, 'commcare_version': None, 'doc_type': 'LastSubmission'}], 'last_submission_for_user': {'app_id': '026c0353a410eef9bed62b128401cc36', 'submission_date': '2020-03-03T17:03:58.706160Z', 'build_id': None, 'device_id': 'Formplayer', 'build_version': None, 'commcare_version': None, 'doc_type': 'LastSubmission'}, 'last_syncs': [{'app_id': None, 'sync_date': '2020-02-27T16:02:25.380522Z', 'build_version': None, 'doc_type': 'LastSync'}], 'last_sync_for_user': {'app_id': None, 'sync_date': '2020-02-27T16:02:25.380522Z', 'build_version': None, 'doc_type': 'LastSync'}, 'last_builds': [], 'last_build_for_user': {'app_id': None, 'build_profile_id': None, 'build_version': None, 'build_version_date': None, 'doc_type': 'LastBuild'}, 'doc_type': 'ReportingMetadata'}, 'username': 'user@example.com', 'first_name': '', 'last_name': '', 'email': '', 'password': 'sha1$c0ffee', 'is_staff': True, 'is_active': True, 'is_superuser': True, 'last_login': '2020-02-27T10:48:22.736319Z', 'date_joined': '2020-02-27T10:42:42.985467Z', 'CURRENT_VERSION': '3.0', 'eulas': [], 'domains': ['nh'], 'domain_memberships': [{'domain': 'nh', 'timezone': 'UTC', 'is_admin': True, 'override_global_tz': False, 'role_id': None, 'location_id': None, 'assigned_location_ids': [], 'program_id': None, 'last_accessed': '2020-03-04', 'doc_type': 'DomainMembership'}]}
```
where `couch_change` is the JSON of a WebUser.

Reply:
I don't understand how the `couch_change` object could be that. Which pillow was this from?